### PR TITLE
LSCP-B

### PR DESCRIPTION
**Pull request by erinrolson**
Implements LSCPB class

### CODE REVIEW

File: spopt/locate/coverage.py
Comment (jGaboardi) [inline spopt/locate/coverage.py:480]: It might be useful to store the LSCP objective value as a class attribute. Then it can be use later in the model *and* kept for info/diagnostics later. Something like the following or thereabouts:

```python
self.lscp_obj_value = lscp.problem.objective.value()
```

**Review by jGaboardi — COMMENTED** [review]
Phenomenal work, @erinrolson! This is a great start here! I think next we need to think about:
* adding in tests for the LSCP-B; and
* creating a notebook demonstrating the LSCP-B

**Comment by jGaboardi** [comment]
cc @qszhao

File: spopt/locate/coverage.py
Comment (jGaboardi) [reply spopt/locate/coverage.py:371]: Look into this later.

File: spopt/locate/coverage.py
Comment (erinrolson) [reply spopt/locate/coverage.py:486]: @jGaboardi I updated the LSCPB class to include this attribute. I also updated the from_cost_matrix class method to set this attribute. However, it seems my implementation isn't quite in line with the package style and I'd like some feedback. Thanks!

Comment (jGaboardi) [reply spopt/locate/coverage.py:486]: hmmm. I see what you mean. Maybe we can (1) declare the `lscp_obj_value` variable within `from_cost_matrix`; (2) pass it into `LSCPB`; assign it as an attribute to the new 


I think `lscp_obj_value` does not need to be set within `LSCPB`, but can be set directly after instantiation. This is because it is not used until we are adding the facility constraint. For example:

```python
lscpb = LSCPB(name, model)
lscpb.lscp_obj_value = lscp.problem.objective.value()

...

FacilityModelBuilder.add_facility_constraint(
    lscpb, lscpb.problem, lscpb.lscp_obj_value
)
```

**Comment by jGaboardi** [comment]
@erinrolson Do we have a basic notebook in this PR that demonstrates a toy problem solving the LSCP-B like we have for the other models, for example [LSCP](https://github.com/pysal/spopt/blob/main/notebooks/lscp.ipynb). We'll want one of those and in it we can formulate the LSCP-B and reference our formulation as @qszhao mentioned on Wednesday.

**Comment by jGaboardi** [comment]
Time to reopen to the *K means heuristic test failure* [issue](https://github.com/pysal/spopt/issues/213)...

**Comment by erinrolson** [comment]
Yes! Working on it. I found a bug in my code for the preselected facility
sites so I was resolving that, updating tests etc. Should be done with the
notebook this afternoon.

**Comment by jGaboardi** [comment]
> Time to reopen to the K means heuristic test failure [issue](https://github.com/pysal/spopt/issues/213)...

I wasn't suggesting this was your fault or pressuring you to try to debug (this is way outside the scope of your GSoC). But if you have the motivation I absolutely won't discourage!

**Comment by jGaboardi** [comment]
@erinrolson I think after removing the `.coverage` file from this PR it will be ready for approval.

**Review by jGaboardi — APPROVED** [review]

**Review by jGaboardi — APPROVED** [review]

**Review by gegen07 — CHANGES_REQUESTED** [review]
- Could you run the lscpb notebook from the beginning? It starts from 30th cell.
- Maybe we can add the `plot_results` function to `util` file? We are using it in all notebooks. @jGaboardi  what do you think?

@erinrolson just small changes to wrap up this awesome work!

**Comment by jGaboardi** [comment]
> Maybe we can add the plot_results function to util file? We are using it in all notebooks. @jGaboardi what do you think?

@gegen07 This could be a nice idea (though not for this PR). We may also discuss integrating viz functionality with `splot`.

File: spopt/locate/coverage.py
Comment (jGaboardi) [reply spopt/locate/coverage.py:361]: @gegen07 Good question. We may have overlooked this...

Comment (gegen07) [reply spopt/locate/coverage.py:361]: Yes!

**Review by jGaboardi — APPROVED** [review]

File: spopt/locate/coverage.py
Comment (jGaboardi) [reply spopt/locate/coverage.py]: @gegen07 This has been addressed by @erinrolson in [1b61f87](https://github.com/pysal/spopt/pull/259/commits/1b61f875b1373b1bfe662933025271dba25dfe03). Once we get your approval then this is good for merging!

**Review by gegen07 — APPROVED** [review]

**Comment by gegen07** [comment]
Thanks @erinrolson for your awesome contribution!! :tada:

**Pull request merged**
